ipykernel/kernelbase.py

```diff
@@ -899,6 +893,45 @@ def _input_request(self, prompt, ident, parent, password=False):
             raise EOFError
         return value
 
+    def _wait_input_request_reply(self):
+        """Wait for an input request reply.
+
+        Raises
+        ------
+        KeyboardInterrupt if a keyboard interrupt is recieved.
+        """
+        # Await a response.
+        reply = None
+        while reply is None:
+            try:
+                ident, reply = self.session.recv(
+                    self.stdin_socket, zmq.NOBLOCK)
+                if not reply:
+                    time.sleep(0.01)
```

Comment:
why the sleep?

Reply:
https://github.com/matplotlib/matplotlib/blob/ec2046f1253583a0eaee9617357b1e82bf41d0b0/lib/matplotlib/backend_bases.py#L2264

Reply:
Because I don't want to use 100% of the processor while waiting for the input. the 0.01 comes from matplotlib: https://github.com/matplotlib/matplotlib/blob/ec2046f1253583a0eaee9617357b1e82bf41d0b0/lib/matplotlib/backend_bases.py#L2264

Reply:
Is there anyway to push this timeout into the zmq recv logic?

Reply:
http://api.zeromq.org/2-1:zmq-recv
There doesn't appear to be a way to do that